Description: fix old-style Haskell imports
 Old-style imports (plain 'System') don't work anymore with GHC 7.4. Newer
 upstreams already fix this, but it hasn't been backported to 2.5.
Author: Iustin Pop <iustin@debian.org>
Forwarded: not-needed
Last-Update: 2012-05-06
--- a/htools/Ganeti/HTools/CLI.hs
+++ b/htools/Ganeti/HTools/CLI.hs
@@ -83,7 +83,7 @@
 import System.Console.GetOpt
 import System.IO
 import System.Info
-import System
+import System.Exit
 import Text.Printf (printf)
 
 import qualified Ganeti.HTools.Version as Version(version)
--- a/htools/Ganeti/HTools/ExtLoader.hs
+++ b/htools/Ganeti/HTools/ExtLoader.hs
@@ -37,7 +37,7 @@
 import Data.Maybe (isJust, fromJust)
 import System.FilePath
 import System.IO
-import System
+import System.Exit
 import Text.Printf (hPrintf)
 
 import qualified Ganeti.HTools.Luxi as Luxi
--- a/htools/Ganeti/HTools/Program/Hspace.hs
+++ b/htools/Ganeti/HTools/Program/Hspace.hs
@@ -31,9 +31,9 @@
 import Data.List
 import Data.Maybe (isJust, fromJust)
 import Data.Ord (comparing)
-import System (exitWith, ExitCode(..))
+import System.Exit (exitWith, ExitCode(..))
 import System.IO
-import qualified System
+import System.Environment (getArgs)
 
 import Text.Printf (printf, hPrintf)
 
@@ -347,7 +347,7 @@
 -- | Main function.
 main :: IO ()
 main = do
-  cmd_args <- System.getArgs
+  cmd_args <- getArgs
   (opts, args) <- parseOpts cmd_args "hspace" options
 
   unless (null args) $ do
--- a/htools/htools.hs
+++ b/htools/htools.hs
@@ -26,7 +26,8 @@
 module Main (main) where
 
 import Data.Char (toLower)
-import System
+import System.Environment
+import System.Exit
 import System.IO
 
 import Ganeti.HTools.Utils
--- a/htools/Ganeti/HTools/Program/Hail.hs
+++ b/htools/Ganeti/HTools/Program/Hail.hs
@@ -27,7 +27,7 @@
 
 import Control.Monad
 import System.IO
-import qualified System
+import System.Environment (getArgs)
 
 import qualified Ganeti.HTools.Cluster as Cluster
 
@@ -51,7 +51,7 @@
 -- | Main function.
 main :: IO ()
 main = do
-  cmd_args <- System.getArgs
+  cmd_args <- getArgs
   (opts, args) <- parseOpts cmd_args "hail" options
 
   let shownodes = optShowNodes opts
--- a/htools/Ganeti/HTools/Program/Hbal.hs
+++ b/htools/Ganeti/HTools/Program/Hbal.hs
@@ -31,11 +31,11 @@
 import Data.List
 import Data.Maybe (isJust, isNothing, fromJust)
 import Data.IORef
-import System (exitWith, ExitCode(..))
+import System.Exit (exitWith, ExitCode(..))
 import System.IO
 import System.Posix.Process
 import System.Posix.Signals
-import qualified System
+import System.Environment (getArgs)
 
 import Text.Printf (printf, hPrintf)
 
@@ -227,7 +227,7 @@
 -- | Main function.
 main :: IO ()
 main = do
-  cmd_args <- System.getArgs
+  cmd_args <- getArgs
   (opts, args) <- parseOpts cmd_args "hbal" options
 
   unless (null args) $ do
--- a/htools/Ganeti/HTools/Program/Hscan.hs
+++ b/htools/Ganeti/HTools/Program/Hscan.hs
@@ -27,10 +27,10 @@
 
 import Control.Monad
 import Data.Maybe (isJust, fromJust, fromMaybe)
-import System (exitWith, ExitCode(..))
+import System.Exit (exitWith, ExitCode(..))
 import System.IO
 import System.FilePath
-import qualified System
+import System.Environment (getArgs)
 
 import Text.Printf (printf)
 
@@ -132,7 +132,7 @@
 -- | Main function.
 main :: IO ()
 main = do
-  cmd_args <- System.getArgs
+  cmd_args <- getArgs
   (opts, clusters) <- parseOpts cmd_args "hscan" options
   let local = "LOCAL"
 
--- a/htools/Ganeti/HTools/IAlloc.hs
+++ b/htools/Ganeti/HTools/IAlloc.hs
@@ -34,7 +34,7 @@
 import Control.Monad
 import Text.JSON (JSObject, JSValue(JSArray),
                   makeObj, encodeStrict, decodeStrict, fromJSObject, showJSON)
-import System (exitWith, ExitCode(..))
+import System.Exit (exitWith, ExitCode(..))
 import System.IO
 
 import qualified Ganeti.HTools.Cluster as Cluster
--- a/htools/test.hs
+++ b/htools/test.hs
@@ -30,7 +30,7 @@
 import System.Console.GetOpt
 import System.IO
 import System.Exit
-import System (getArgs)
+import System.Environment (getArgs)
 import Text.Printf
 
 import Ganeti.HTools.QC
